clang/lib/AST/ASTContext.cpp

```diff
@@ -5216,6 +5230,38 @@ QualType ASTContext::getDependentBitIntType(bool IsUnsigned,
   return QualType(New, 0);
 }
 
+QualType
+ASTContext::getPredefinedSugarType(PredefinedSugarType::Kind KD) const {
+  llvm::FoldingSetNodeID ID;
+  PredefinedSugarType::Profile(ID, llvm::to_underlying(KD));
+
+  void *InsertPos = nullptr;
+  if (PredefinedSugarType *Existing =
+          PredefinedSugarTypes.FindNodeOrInsertPos(ID, InsertPos))
+    return QualType(Existing, 0);
```

Comment:
I mentioned this in another comment, but it's a waste to use a hash table to unique a set with only 3 possible elements. You could just use an array instead.